examples/withdraw.py

```diff
@@ -0,0 +1,161 @@
+    )
+
+    calls = [
+        paraclear_contract.functions["withdraw"].prepare(
+            token_address=hex_to_int(usdc_address),
+            amount=amount * 10**paraclear_decimals,
+        ),
+        l2_bridge_contract.functions["initiate_withdraw"].prepare(
+            l1_recipient=hex_to_int(l1_recipient),
+            amount=amount * 10**usdc_decimals,
+        ),
+    ]
+    withdraw_info = await account.execute(calls=calls, max_fee=get_random_max_fee())
+    withdraw_tx_hash = hex(withdraw_info.transaction_hash)
+    logging.info(f"Waiting for withdraw to complete: {withdraw_tx_hash}")
+    tx_status = await account.client.wait_for_tx(
+        tx_hash=withdraw_info.transaction_hash,
+        wait_for_accept=True,
+    )
+    logging.info(f"L2 withdraw completed: {tx_status}")
+
+    # Check balance
+    usdc_contract = await Contract.from_address(
+        provider=account, address=usdc_address, proxy_config=get_proxy_config()
+    )
+    usdc_bal = await usdc_contract.functions["balanceOf"].call(account=account.address)
+    logging.info(f"USDC L2 balance is {usdc_bal.balance / 10**usdc_decimals}")
+
+    return account.client, withdraw_tx_hash
+
+
+async def withdraw_from_l1_bridge(
+    l1_recipient: str, amount: int, config: Dict, w3: Web3
+) -> None:
+    logging.info("Withdraw from L1 token bridge contract to L1 recipient")
+
+    l1_bridge_address = config["bridged_tokens"][0]["l1_bridge_address"]
+    l1_bridge_abi = json.load(open(os.path.abspath("./abis/l1_bridge_abi.json")))
+    usdc_decimals = config["bridged_tokens"][0]["decimals"]
+
+    l1_contract = w3.eth.contract(address=l1_bridge_address, abi=l1_bridge_abi)
+
+    nonce = w3.eth.get_transaction_count(l1_recipient, "pending")
+    tx = {
+        "maxFeePerGas": w3.toWei("2", "gwei"),
+        "maxPriorityFeePerGas": w3.toWei("1", "gwei"),
+        "nonce": nonce,
+        "type": 2,
+    }
+    withdraw_tx = l1_contract.functions.withdraw(
+        amount * 10**usdc_decimals, l1_recipient
+    ).build_transaction(tx)
+    tx_hash = w3.eth.send_transaction(withdraw_tx)
+    logging.info(f"L1 withdraw tx hash: { HexBytes(tx_hash).hex()}")
+
+
+# Primary Coroutine
+async def main(eth_private_key_hex: str) -> None:
+    w3, eth_account = get_l1_eth_account(eth_private_key_hex)
+
+    # Load Paradex config
+    paradex_config = await get_paradex_config()
+
+    # Generate Paradex account (only local)
+    paradex_account_address, paradex_account_private_key_hex = generate_paradex_account(
+        paradex_config, eth_account.key.hex()
+    )
+
+    # Starknet account
+    account = get_account(
+        paradex_account_address, paradex_account_private_key_hex, paradex_config
+    )
+
+    amount = 1  # 1 USDC
+
+    # This method only waits for `ACCEPTED_ON_L2` status
+    # Status change to `ACCEPTED_ON_L1` could take up to 12 hours
+    client, withdraw_tx_hash = await withdraw_from_paraclear(
```

Comment:
this implies that user already has funds on Paraclear
Suggestion - we either move it in try..catch or we first check whether Paraclear has withdrawable balance and exit if user doesn't have balance

Reply:
Yes, this code sample assumes user has funds on Paraclear that they want to withdraw.